src/screens/MainApp/MyDiagnoses/index.js

```diff
@@ -159,33 +213,46 @@ const MyDiagnoses = ({ navigation }) => {
 
   return (
     <Background>
-      <HeaderForScrolling show={scrolling} />
-      {diagnoses &&
-        <SwipeListView
-          ref={flatListRef}
-          data={diagnoses}
-          showsVerticalScrollIndicator={false}
-          style={styles.container}
-          ListHeaderComponent={<StaticHeader />}
-          renderItem={({ item }) => <RenderDiagnose animationDuration={DIAGNOSE_ANIMATION_DURATION} item={item} />}
-          keyExtractor={item => item.id}
-          disableVirtualization={false}
-          contentContainerStyle={styles.flatListContainer}
-          onScroll={({ nativeEvent }) => { setScrolling(isScrollWithinSnapThreshold(nativeEvent)) }}
-          renderHiddenItem={(data, rowMap) => (
-            <View style={[styles.hiddenField]}>
-              <SvgButton
-                height={verticalScale(30)}
-                width={scale(21)}
-                svg={trashIcon}
-              />
-            </View>
-          )}
-          disableRightSwipe
-          disableLeftSwipe={!isListSwipeEnabled}
-          onRowOpen={handleListItemSwipe}
-          rightOpenValue={scale(-375)}
-        />}
+      <SwipeListView
+        ref={flatListRef}
+        data={diagnoses}
+        extraData={tab}
+        showsVerticalScrollIndicator={false}
+        style={styles.container}
+        ListHeaderComponent={
```

Comment:
I tried to put all this in a arrow function that returned the static header and the tab-view all together but it changed the tab speed when changing tabs @nahudalla  @MPucheta

Reply:
Mmm, the best way is what you described, but also moving this in a `useMemo` hook, to provide a `memoized` value that changes according to dependencies (otherwise each render could be creating those components).

Something like this 

![image](https://user-images.githubusercontent.com/35865469/81564955-ef60b700-936e-11ea-8772-609712060668.png)

If the code you submitted works okay/fast do not worry then.

Reply:
> I tried to put all this in a arrow function that returned the static header and the tab-view all together but it changed the tab speed when changing tabs @nahudalla @MPucheta

Probably you should have used a memoized arrow function, with `useMemo` or `useCallback`. Anyway, if this works, it's not that bad to pass rendered components as a prop.